src/Microsoft.AspNetCore.HttpsPolicy/HttpsPolicyOptions.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.HttpsPolicy
+{
+    /// <summary>
+    /// Options for the HttpsPolicyMiddleware middleware
+    /// </summary>
+    public class HttpsPolicyOptions
+    {
+        /// <summary>
+        /// Initializes a new <see cref="HttpsPolicyOptions"/>
+        /// </summary>
+        public HttpsPolicyOptions()
+        {
+            HstsOptions = new HstsOptions();
+        }
+
+        /// <summary>
+        /// Whether to use HTTP Strict-Transport-Security (HSTS) on all HTTPS requests.
+        /// </summary>
+        public bool SetHsts { get; set; }
+
+        /// <summary>
+        ///  Options for using HSTS
+        /// </summary>
+        public HstsOptions HstsOptions { get; set; } // TODO should this be initialized at all? We could remove the bool if so.
+
+        /// <summary>
+        /// The status code to be used for Url Redirection
+        /// </summary>
+        public int StatusCode { get; set; } = StatusCodes.Status301MovedPermanently; // TODO throw ArgumentOutOfRangeException from UrlRewrite redirect rule?
```

Comment:
It's ok as it is. I don't think we need to over protect against a non redirect status code. I would change the property name and the doc comment to reflect that is a redirect status code. Property would be RedirectStatusCode. What do you think?

Reply:
Yes RedirectStatusCode.